### PR TITLE
feat:(feedback-api-layer) - Created the API Communication Layer package for the feedback

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 839, 865

# Explain the feature
NPM package for the One Platform which enables the feedback to talk to backend apis.

## Does this PR introduce a breaking change
Yes

Migration Path. 
1) Once this is merged Configure SPAs with new feedback package and api layer from one platform. and clean the SSI.
## Screenshots

<details>
<summary>View Screenshots</summary>

image

</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/feedback-api-layer/src/feedback.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    xhr.setRequestHeader('Accept', 'application/json');
+    xhr.setRequestHeader("Authorization", authToken || `${(window.OpAuthHelper.jwtToken) ? `Bearer ${window.OpAuthHelper.jwtToken}` : null}`);
+    xhr.send(feedbackQuery);
+
+    xhr.addEventListener("readystatechange", function () {
+        if (this.readyState === 4) {
+            let response = JSON.parse(this.responseText);
+            if (response.data.createFeedback) {
+                (window.OpNotification) ? window.OpNotification.success({ subject: `Submitted Feedback` }) : alert('Submitted Feedback')
```
Comment (ghanlohar) [inline packages/feedback-api-layer/src/feedback.js:43]:
```suggestion
                (window.OpNotification) ? window.OpNotification.success({ subject: `Your feedback has been submitted successfully` }) : alert('Your feedback has been submitted successfully')
```

**Review by ghanlohar — APPROVED** [review]
Rest looks good to me.

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]
This was supposed to go in the ssi-service package.

Also create a html file similar to `feedback.slim.html` that will take care of adding the script in the DOM. That will allow anyone to add this script to their app by just adding an SSI include script.

I think this needs to be improved a lot, but due to limited resources right now, just move it inside ssi, and make the changes mentioned to use it for the dependent apps.

We will have to come back to this later.

**Comment by riginoommen** [comment]
The goal of this layer is to seperate the feedback dependencies as seperate one once this is approved this will be published as seperate github package. As this is an opensource project we need to provide the plug in support for everything that we uses. as from web component this will be the medium of sending the data to backend.

**Review by ghanlohar — APPROVED** [review]

File: packages/feedback-api-layer/src/feedback.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+            } else {
+                (window.OpNotification) ? window.OpNotification.danger({ subject: `Error in Feedback Submission` }) : alert('Error in Feedback Submission')
+            }
+        }
+    });
+}
+
+module.exports.sendFeedback = this.sendFeedback;
```
Comment (deshmukhmayur) [inline packages/feedback-api-layer/src/feedback.js:51]:
```suggestion
module.exports.sendFeedback = this.sendFeedback;

```

File: packages/feedback-api-layer/test/feedback.test.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * Last modified  : 2021-01-25 11:38:41
+ */
+
+const sendFeedback = require('../src/feedback');
+
+describe('feedback-api-layer', () => {
+    it('sendFeedback', async () => {
+        expect(sendFeedback).toBeDefined();
+    });
+});
```
Comment (deshmukhmayur) [inline packages/feedback-api-layer/test/feedback.test.ts:26]:
```suggestion
});

```

File: packages/feedback-api-layer/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    },
+    plugins: [],
+    resolve: {
+        extensions: [ '.tsx', '.ts', '.js' ],
+    },
+    output: {
+        filename: 'feedback.js',
+        path: path.resolve(__dirname, 'dist'),
+    },
+};
```
Comment (deshmukhmayur) [inline packages/feedback-api-layer/webpack.config.js:42]:
```suggestion
};

```

File: packages/feedback-api-layer/jest.config.js [hunk 5]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    setupFiles: ['./dist/feedback.js'],
+};
```
Comment (deshmukhmayur) [inline packages/feedback-api-layer/jest.config.js:3]:
```suggestion
};

```

**Review by deshmukhmayur — CHANGES_REQUESTED** [review]
Is this PR ready for review? As per our last discussion, some changes were pending due to another PR in op-component-library, is it still being worked on?

**Pull request closed without merging**